halo2-base/README.md

```diff
@@ -0,0 +1,498 @@
+
+    pub selector: Vec<bool>,
+
+    pub advice_equality_constraints: Vec<(ContextCell, ContextCell)>,
+    
+    pub constant_equality_constraints: Vec<(F, ContextCell)>,
+}
+```
+The `witness_gen_only` is set for witness computation, such as during Mock proving. Otherwise it is set to false.
+
+The `Context` holds all equality and constant constraints as a `Vec` of `ContextCell` tuples representing the positions of the two cells to constrain. `advice`, `selector`, store the respective columns values of the `Context`'s section of the two columns. `cells_to_lookup` tracks `AssignedValue`'s of cells marked for use in a lookup table within the `Context`.
+
+[ContextCell](https://github.com/axiom-crypto/halo2-lib/blob/release-0.3.0/halo2-base/src/lib.rs#L94) is a pointer to a specific cell within a `Context` identified by the Context's `context_id` and the cells relative `offset` from the first cell of the advice column of the Context.
+
+``` rust ignore
+#[derive(Clone, Copy, Debug)]
+pub struct ContextCell {
+    /// Identifier of the [Context] that this cell belongs to.
+    pub context_id: usize,
+    /// Relative offset of the cell within this [Context] advice column.
+    pub offset: usize,
+}
+```
+
+[AssignedValue](https://github.com/axiom-crypto/halo2-lib/blob/release-0.3.0/halo2-base/src/lib.rs#L105) represents a specific `Assigned` value assigned to a cell within a specific `Context` of a circuit referenced by a `ContextCell`. 
+```rust ignore
+pub struct AssignedValue<F: ScalarField> {
+    pub value: Assigned<F>,
+
+    pub cell: Option<ContextCell>,
+}
+```
+    
+[Assigned](https://github.com/zcash/halo2/blob/main/halo2_proofs/src/plonk/assigned.rs#L11) is a wrapper enum for values assigned to a cell within a circuit which stores the value as a fraction and marks it for batched inversion using Montgomery's trick https://zcash.github.io/halo2/background/fields.html#montgomerys-trick. Performing batched inversion allows for the computation of the inverse of all marked values with a single inversion operation.
+```rust ignore
+pub enum Assigned<F> {
+    /// The field element zero.
+    Zero,
+    /// A value that does not require inversion to evaluate.
+    Trivial(F),
+    /// A value stored as a fraction to enable batch inversion.
+    Rational(F, F),
+}
+```
+    
+<br>
+
+## [QuantumCell](https://github.com/axiom-crypto/halo2-lib/blob/release-0.3.0/halo2-base/src/lib.rs#L53)
+
+QuantumCell is a helper enum that abstracts the scenarios in which a value is assigned to the advice column in Halo2-base. Assigning existing or constant values to the advice column requires manually specifying the enforeced constraints on top of assigning the value leading to bloated code. QuantumCell handles these technical operations for you, all a developer needs to do is specify which enum option in `QuantumCell` there value corresponds to. The `assign_cell()` method `Context` then handles the technical operations for you.
```

Comment:
add: "Without QuantumCell, assigning existing or constant values..." 

enforced - typo

for you -- all a developer needs

there value ->their value

Reply:
`assign_cell()` method **in** `Context`?